subprojects/packagefiles/libyaml/meson.build

```diff
@@ -0,0 +1,54 @@
+project(
+  'yaml-0.1', 'c',
+  license: 'MIT',
+  meson_version: '>=0.49.2',
+  version: '0.2.5',
+)
+
+pkg = import('pkgconfig')
+
+add_project_arguments('-DHAVE_CONFIG_H=1', language: 'c')
+
+yaml_version = meson.project_version()
+yaml_major_version = yaml_version.split('.')[0]
+yaml_minor_version = yaml_version.split('.')[1]
+yaml_patch_version = yaml_version.split('.')[2]
+cdata = configuration_data({
+  'YAML_VERSION_MAJOR' : yaml_major_version,
+  'YAML_VERSION_MINOR' : yaml_minor_version,
+  'YAML_VERSION_PATCH' : yaml_patch_version,
+})
+cdata.set_quoted('YAML_VERSION_STRING', meson.project_version())
+
+config_h = configure_file(
+  output: 'config.h',
+  configuration: cdata)
+
+yaml_lib = library('yaml',
+  'src/api.c',
+  'src/dumper.c',
+  'src/emitter.c',
+  'src/loader.c',
+  'src/parser.c',
+  'src/reader.c',
+  'src/scanner.c',
+  'src/writer.c',
+  include_directories: [
+    include_directories('.'),
+    ],
+  version: meson.project_version(),
+  soversion: yaml_major_version + '.' + yaml_minor_version,
```

Comment:
Upstream library versioning uses libtool compatible versioning `-version-info 2:9:0` which translates to `version: '2.0.9'` and the soversion is directly inferred from version as `soversion: '2'`. it does not include multiple components.